Derive Debug on public structs

Thomasdezeeuw opened the pull request:
And add the `warn(missing_debug_implementations)` attribute.

carllerche commented on src/net/tcp.rs:
```diff
@@ -208,12 +209,6 @@ impl TcpListener {
     }
 }
 
-impl fmt::Debug for TcpListener {
```
Could you explain why you changed this implementation? The existing implementation is a bit cleaner.

Thomasdezeeuw replied:
Well now the contents of the struct is not really being displayed, which can be deceiving when debugging problems. Just using derive is just as simple, while actually display the contents. Is there a reason this was written by hand?

carllerche replied:
IMO the following isn't super useful:

```rust
TcpStream { inner: TcpStream { inner: TcpStream } }
```

I believe that it was written by hand to avoid this sort of output.

carllerche commented on src/net/tcp.rs:
```diff
@@ -624,12 +622,6 @@ impl<'a> AsyncWrite for &'a TcpStream {
     }
 }
 
-impl fmt::Debug for TcpStream {
```
Same question as above.

carllerche commented on src/net/udp/frame.rs:
```diff
@@ -116,7 +117,7 @@ impl<C: UdpCodec> Sink for UdpFramed<C> {
             Ok(Async::Ready(()))
         } else {
             Err(io::Error::new(io::ErrorKind::Other,
-                               "failed to write entire datagram to socket"))
```
This seems to be an unrelated line to the commit.

carllerche commented on src/net/udp/mod.rs:
```diff
@@ -373,16 +373,11 @@ impl UdpSocket {
     }
 }
 
-impl fmt::Debug for UdpSocket {
```
Same as above.

Thomasdezeeuw commented:
Something has gone wrong with the commit and now changes are double, I'll fix that shortly.

Thomasdezeeuw commented:
Closing this in favor of https://github.com/tokio-rs/tokio/pull/62, this branch got messed up. But I've addresses all your comments @carllerche. In the new PR I only added the warning and derived Debug on struct that didn't have an implementation yet.
